src/mem/heap/heap_simple.c

```diff
@@ -93,10 +93,14 @@ void free(void *ptr) {
 }
 
 void *malloc(size_t size) {
+	return memalign(1, size);
```

Comment:
Great! But aligning memory to the 1-byte boundary is not a correct way, since 

```
The address of the allocated memory will be a multiple of alignment, which
       must be a power of two and a multiple of sizeof(void *)
```

you can read about this here https://linux.die.net/man/3/memalign

It's very usual behavior when the returned address is word aligned (word size can be 4 or 8 bytes).

Reply:
So why is this wrong? 1 is power of 2 and when we pass 1 (I added one more simple fix to check size condition) we do it like simple malloc.

Reply:
But 1 is not a multiple of sizeof(void *), isn't it?

Reply:
Sorry, I misunderstood this sentence. So we should pass sizeof (void *) in malloc? And additionally check this condition in memalign?

Reply:
Yes, thank you! But it's required only for malloc() to be aligned.

Reply:
So it was my mistake, I told you that is required by memalign, but it's required by malloc(). So we should pass this alignment (sizeof (void *) to memalign only to implement malloc).

Reply:
Done :).